quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -124,10 +127,11 @@ class
   precondition _ _ = True
 
   -- | Postcondition on the `a` value produced at some step.
-  -- The result is `assert`ed and will make the property fail should it be `False`. This is useful
-  -- to check the implementation produces expected values.
-  postcondition :: state -> Action state a -> LookUp -> a -> Bool
-  postcondition _ _ _ _ = True
+  --
+  -- When 'postcondition' returns @Just err@, the property will fail, and @err@ will be shown as a
+  -- counter-example. This is useful to check th implementation produces expected values.
+  postcondition :: state -> Action state a -> LookUp -> a -> Postcondition
```

Comment:
I would be much happier with a `Assertable` type class here (just like how QuickCheck has `Testable`). That way you can return a boolean if you want.

Reply:
Also, the comment is now obsolete.